hal/ia32/cpu.c

```diff
@@ -605,3 +606,25 @@ void _hal_cpuInit(void)
 //	hal_cpuDebugGuard(1, 3);
 #endif
 }
+
+
+void hal_cpuTlsSet(tls_t *tls, cpu_context_t *ctx)
+{
+	/* At the end of TLS there must be pointer to itself */
+	*(ptr_t *)((ptr_t)tls->tls_base + tls->tdata_sz + tls->tbss_sz) = (ptr_t)tls->tls_base + tls->tdata_sz + tls->tbss_sz;
+
+	ctx->gs = SEL_TLS;
+	u32 base = (u32)((char *)tls->tls_base + tls->tbss_sz + tls->tdata_sz);
+	u32 limit = VADDR_KERNEL;
```

Comment:
These should be of `ptr_t` type

Reply:
I think descriptors should be also, I guess these also scale up to 64 bits on x64, but I'm not sure